Firebend.AutoCrud.EntityFramework/Abstractions/Client/EntityFrameworkUpdateClient.cs

```diff
@@ -60,6 +60,9 @@ public abstract class EntityFrameworkUpdateClient<TKey, TEntity> : AbstractDbCon
             _isDefaultPublisher = domainEventPublisher is DefaultEntityDomainEventPublisher;
         }
 
+        protected virtual Task<TEntity> OnBeforeAddAsync(IDbContext context, TEntity entity, IEntityTransaction transaction, CancellationToken cancellationToken)
+            => Task.FromResult(entity);
```

Comment:
while we are in here, should we also provision an `OnBeforeUpdateAsync` function? 
might be nice to have it in the future

Reply:
added one for patch and bumped the versions